pkg/acsengine/params.go

```diff
@@ -173,6 +173,9 @@ func getParameters(cs *api.ContainerService, generatorCode string, acsengineVers
 	for _, agentProfile := range properties.AgentPoolProfiles {
 		addValue(parametersMap, fmt.Sprintf("%sCount", agentProfile.Name), agentProfile.Count)
 		addValue(parametersMap, fmt.Sprintf("%sVMSize", agentProfile.Name), agentProfile.VMSize)
+		if agentProfile.AvailabilityZones != nil && len(agentProfile.AvailabilityZones) > 0 {
```

Comment:
...and then we just call `hasAvailabilityZones` from here

(sorry, the first part of the elipsis is below ;) )

Basically, that way `HasAvailabilityZones` in the params template directly correlates to the criteria here. We always want to define the param, and then inject it using this `addValue` function, according to the exact same criteria.